File: ABC_Chapter11_Exercise1/app/src/main/java/com/example/cyu/abc_chapter11_exercise1/MainActivity.java
package com.example.cyu.abc_chapter11_exercise1;

import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    int weight;
    int height;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText txtWeight = findViewById(R.id.txtWeight);
        final EditText txtHeight = findViewById(R.id.txtHeight);

        final SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);

        Button btnCompute = findViewById(R.id.btnCompute);
        btnCompute.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                weight = Integer.parseInt(txtWeight.getText().toString());
                height = Integer.parseInt(txtHeight.getText().toString());

                SharedPreferences.Editor editor = sharedPref.edit();

                editor.putInt("key1", weight);
                editor.putInt("key2", height);
                editor.commit();

                startActivity(new Intent(MainActivity.this, Payment.class));
            }
        });

    }
}

File: ABC_Chapter11_Exercise1/app/src/main/java/com/example/cyu/abc_chapter11_exercise1/MainActivity.kt
package com.example.cyu.abc_chapter11_exercise1

import android.content.Intent
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    internal var weight: Int = 0
    internal var height: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val txtWeight = findViewById<EditText>(R.id.txtWeight)
        val txtHeight = findViewById<EditText>(R.id.txtHeight)
        val sharedPref = PreferenceManager.getDefaultSharedPreferences(this)
        val btnCompute = findViewById<Button>(R.id.btnCompute)

        btnCompute.setOnClickListener {
            weight = Integer.parseInt(txtWeight.text.toString())
            height = Integer.parseInt(txtHeight.text.toString())

            val editor = sharedPref.edit()

            editor.putInt("key1", weight)
            editor.putInt("key2", height)
            editor.apply()

            startActivity(Intent(this@MainActivity, Payment::class.java))
        }

    }
}